lib/rules/jsx-no-literals.js

```diff
@@ -25,24 +25,38 @@ module.exports = {
       properties: {
         noStrings: {
           type: 'boolean'
+        },
+        validateProps: {
+          type: 'boolean'
         }
       },
       additionalProperties: false
     }]
   },
 
-  create: function(context) {
-    const isNoStrings = context.options[0] ? context.options[0].noStrings : false;
-    const sourceCode = context.getSourceCode();
+  create: function({
+    report,
+    options: [configs = {}],
+    getSourceCode
+  }) {
+    const {
+      noStrings = false,
+      validateProps = false
+    } = configs;
+    const isNoStrings = noStrings;
+    const shouldValidateProps = validateProps;
+    const sourceCode = getSourceCode();
```

Comment:
in particular, it seems that eslint 3 relies on the receiver (the `this` value) of `getSourceCode` being `context`.